File: Java/Data Structures & Algorithms Module - Java/Week10/BSTAttempt2/BSTNode.java
/*
BSTNode.java
Billy Cussen
26/11/2020
*/
public class BSTNode<T>{

    T element;
    BSTNode<T> left;
    BSTNode<T> right;

    public BSTNode(T element){
        this.element = element;
    }

    @Override
    public String toString(){
        String left = this.left == null ? "NULL" : this.left.element.toString();
        String right = this.right == null ? "NULL" : this.right.element.toString();
        return "Current: "+this.element+", Left: "+left+", Right: "+right;
    }

}

File: Python/Data Structures & Algorithms Module - Python/Week10/BST Attempt 3/BSTNode.py
"""
BSTNode.py
Billy Cussen
09/12/2020
"""

class BSTNode:
    def __init__(self, element):
        self.element = element
        self.left = None
        self.right = None

    def toString(self):
        left = "NULL" if self.left == None else str(self.left.element)
        right = "NULL" if self.right == None else str(self.right.element)
        return "Current: "+str(self.element)+", Left: "+left+", Right: "+right
